### PR TITLE
Rename some macros in `hdk_derive` to reflect the fact that they're only really used via the HDI

### PR DESCRIPTION
**Pull request by pdaoust**
### Summary

Renamed `hdk_entry_helper`, `hdk_entry_types`, `hdk_entry_types_conversions`, `hdk_entry_types_name_registration`, `hdk_link_types`, and `hdk_to_coordinates` to `hdi_*`, to reflect the fact that integrity zomes (the main consumer of the aforementioned macros) are the only place these are used.

I'd love to remove the prefix entirely, but I understand why I can't -- it's because there are naming clashes with various bits and bobs.

**NOTE**: I haven't renamed all of the macros -- `hdk_extern` is still `hdk_` -- because they're not specific to the HDI, and you could consider the HDI a subset of the HDK for conceptual purposes, so it could be an appropriate name for use in integrity zomes too.

### TODO:
- [x] CHANGELOGs updated with appropriate info
- [x] All code changes are reflected in docs, including module-level docs

### CODE REVIEW

**Comment by pdaoust** [comment]
crap, didn't notice my `develop` was so out-of-date. I'll merge it in and push

**Comment by pdaoust** [comment]
Closes #3380

File: crates/hdk_derive/CHANGELOG.md
Comment (jost-s) [inline crates/hdk_derive/CHANGELOG.md:1]: We're moving to using only the holochain package Changelog, so the text here should go there.

File: crates/hdk_derive/CHANGELOG.md [hunk 1]
```diff
@@ -7,6 +7,8 @@ The format is based on [Keep a Changelog](https://keepachangelog.com/en/1.0.0/).
 
 ## \[Unreleased\]
 
+- **BREAKING CHANGE**: Renamed `hdi_entry_helper`, `hdi_entry_types`, `hdi_entry_types_conversions`, `hdi_entry_types_name_registration`, `hdk_link_types`, and `hdk_to_coordinates` to `hdi_*`, to reflect the fact that integrity zomes (the main consumer of the aforementioned macros) are the only place these are used. ([#4115](https://github.com/holochain/holochain/pull/4115))
```
Comment (jost-s) [inline crates/hdk_derive/CHANGELOG.md:10]:
```suggestion
- **BREAKING CHANGE**: Renamed `hdk_entry_helper`, `hdk_entry_types`, `hdk_entry_types_conversions`, `hdk_entry_types_name_registration`, `hdk_link_types`, and `hdk_to_coordinates` to `hdi_*`, to reflect the fact that integrity zomes (the main consumer of the aforementioned macros) are the only place these are used. ([#4115](https://github.com/holochain/holochain/pull/4115))
```

File: crates/hdk_derive/CHANGELOG.md [hunk 2]
```diff
@@ -57,7 +59,7 @@ The format is based on [Keep a Changelog](https://keepachangelog.com/en/1.0.0/).
 
 ## 0.3.0-beta-dev.23
 
-- **BREAKING CHANGE**: Renamed macros `hdk_entry_defs` to `hdk_entry_types` and `entry_def` to `entry_type` for naming consistency with `hdk_link_types` [\#2979](https://github.com/holochain/holochain/pull/2979)
+- **BREAKING CHANGE**: Renamed macros `hdi_entry_defs` to `hdi_entry_types` and `entry_def` to `entry_type` for naming consistency with `hdk_link_types` [\#2979](https://github.com/holochain/holochain/pull/2979)
```
Comment (jost-s) [inline crates/hdk_derive/CHANGELOG.md:62]: Past changes shouldn't be modified.
```suggestion
- **BREAKING CHANGE**: Renamed macros `hdk_entry_defs` to `hdk_entry_types` and `entry_def` to `entry_type` for naming consistency with `hdk_link_types` [\#2979](https://github.com/holochain/holochain/pull/2979)
```

File: docs/specs/src/# Comments.md
Comment (jost-s) [inline docs/specs/src/# Comments.md:1]: What is this fiile?

**Review by jost-s — CHANGES_REQUESTED** [review]

File: crates/hdk_derive/CHANGELOG.md (hunk 2 above)
Comment (pdaoust) [reply crates/hdk_derive/CHANGELOG.md:62]: whoops, over-eager global search and replace :D

File: docs/specs/src/# Comments.md
Comment (pdaoust) [reply docs/specs/src/# Comments.md:1]: Mistake -- from another branch where I'm working on the white paper.

File: crates/hdk_derive/CHANGELOG.md
Comment (pdaoust) [reply crates/hdk_derive/CHANGELOG.md:1]: How do I format it? Should I put it under some sort of `hdk_derive` header?

Comment (jost-s) [reply crates/hdk_derive/CHANGELOG.md:1]: There are no set guidelines, but I suggest to start with something like `**BREAKING CHANGE**` and yes, then followed by the HDK as the place where the change happened.

**Review by jost-s — APPROVED** [review]

**Comment by ThetaSinner** [comment]
Would it be okay to hold this change for 0.5? I know it's just naming but it's work that Holo could do without for the launch I think. 

Any different opinions on this?

**Pull request closed without merging**